### DISCUSSION TITLE
Windows: REAPER global shortcuts bug behavior (renaming things triggers them)

### DISCUSSION THREAD

**Discussion by atencion** [helgoboss/helgobox#1053] [Ideas]
Hi! Im having trouble renaming things since i have some trouble renaming things in Playtime 2 since whenever i press spacebar, for instance, my PLAY arrangement global shortcut gets trigggered. I think there is a reaper option that will override shortcuts when you press keys in an input field, and that wourl be great for PLaytime to have the same option. It happens with most shortcuts i have globally mapped, like Solo (S) or mute (M) so i can`t really rename things in Playtime withought messing things up.

Ive tried changing to Prefer Native Pop up menues, but this doesnt change the playtime input input box to reaper native one, so it`s the same.

Hope this description helps, if not, just let me know.

THANKS!

**Comment by helgoboss** [comment]
Hi @atencion. So you added some of your own shortcuts in REAPER's action menu with Scope = "Global" and no modifier key (no ctrl/shift/cmd etc.). And now they are triggered whenever you type something into Playtime's text fields. Is that correct?

**Comment by atencion** [comment]
Exactly. I use a lot of those type of shortcuts with no modifiers, like solo mute and specially the SPACEBAR as global so i can control the mix and check different things no matter in wich plugin i`m in. 

So when i`m renaming something in any input field in PLaytime, the corresponding Action to that pressed key shortcut it`s triggered. I wonder if is there any way to let reaper know i`m in a input field so reaper would ignore them as it happens in other input fields in other plugins. Thanks!

**Comment by helgoboss** [comment]
Interesting, this is a Windows-only issue.

Oh, and by the way. If you have SPACE mapped globally, you can't benefit from Playtime's default transport synchronization mode "Partial". Pretty big disadvantage because then, pressing SPACE within Playtime always triggers REAPER's transport as well, no way to just let Playtime play isolated (except of course clicking the hand/play button).

**Reply by atencion** [reply]
Yeah, i fugured it out finally. hahah and changed the global space shortcut to its default. Playtime is changing my workflow for good and i`m all for it. hahaha  Now the renaming part is different, cause the must and sole shortcuts (M and S) for example i used them a lot and are made global so i can check while mixing and adjusting settings in other plugins. Same with other direct letters or numbers shortcuts withouth modifiers. Renaming in Playtime becomes impossible. THanks!

**Comment by Chianti** [comment]
Same problem (Windows10). I deleted all of Reaper's default shortcuts, and I only add back the Actions I want/need, on keys which make sense for my Spanish keyboard. I have about 30 keys. _All of my shortcuts are global_. At the moment, I'm just renaming tracks in Reaper instead of in Playtime2, and assumed it was just some kind of small bug that would get fixed soon, as I don't think any of my VST plugins react this way with global shortcuts; for instance, when saving preset names within a synth plugin. At least, I haven't noticed it.

**Reply by helgoboss** [reply]
Yea, I'll attempt to fix it. It's not as easy as it sounds, but I think it's possible.

**Reply by atencion** [reply]
What about this experimental mode you have with native reaper or system menues and you keep in that with for input boxes in PLaytime as well?  Dont know, could be a quick fix till something else comes up and we can use that.

Thanks again for your work!

**Reply by helgoboss** [reply]
That sounds like even more effort, so it can't serve as a quick fix.

It's possible that I will release 2.16.1 without this shortcut fix in order to not delay it. But it will be the next thing after that.

**Reply by helgoboss** [reply]
There's even another possibility: I could disable global REAPER shortcuts when the Playtime window is focused. Otherwise, Playtime has some single-letter shortcuts itself that wouldn't work when those shortcuts are global REAPER shortcuts, e.g. SPACE (especially SPACE). Might need to make this configurable.

**Reply by atencion** [reply]
Yeah, that could be. I^m sure there will be a way to implement it. Don`t worry. In my case at least is not that urgent now i disabled SPACEBAR as global. I`m not sure about other users. I rename things other way in the meantime. Mostly writing outside reaper, copying and pasting in playtime. hahaha If there is a will there is a way.

**Comment by helgoboss** [comment]
I created issue #1083 from this discussion. It's clear that this needs to be done. And since it's different in macOS and Windows right now, it can also be counted as a bug.

**Comment by helgoboss** [comment]
Ha, found a way to fix this on Windows. The easier way is to ignore global hotkeys in general when the Helgobox App window is focused. But only ignoring them when in a text field should not be much more effort, I will try.

This didn't make it into 2.16.1, which I published today. But it will be one of the first things making it into 2.16.2.

**Comment by helgoboss** [comment]
I realized that there's another hotkey-related difference between macOS and Windows. On macOS, non-Playtime shortcuts are forwarded to REAPER, in Windows not. Actually, I like the behavior on Windows more.

Here's my target picture concerning the "Helgobox App vs. REAPER shortcuts" situation, for both Windows and macOS:

### REAPER shortcuts defined with scope "Global" should work, even if the Helgobox App window is focused

Reasoning: This is the whole purpose of the "Global" scope. If the user wanted a shortcut to work only within the REAPER main window, he wouldn't need to declare it as "Global" in the first place.

Current situation:

- macOS: ✅ 
- Windows: ✅ 

### Text fields should be an exception to above rule (except if scope is "Global + text fields")

Current situation:

- macOS: ✅ 
- Windows: 🚫

### REAPER shortcuts not defined with scope "Global" should not work when the Helgobox App window is focused

Reasoning: While a "pass-through" of shortcuts unused by Playtime would sound convenient in some cases, it's not future-proof. People would get used to it. But suddenly Playtime itself would get a shortcut and then it would suddenly trigger something else - principle of least surprise violated.

Current situation:

- macOS: 🚫 
- Windows: ✅

**Comment by helgoboss** [comment]
Fixed. The mentioned fixes/changes will be in upcoming 2.16.2.
